File: 5.8/ts/reverseDigits.js
function enforceInteger(n) {
    return n - (n % 1);
}
function reverseIntegerDigits(n) {
    n = enforceInteger(n);
    var reversed = n % 10;
    n = (n - reversed) / 10;
    while (n != 0) {
        var digit = n % 10;
        n = (n - digit) / 10;
        reversed = (reversed * 10) + digit;
    }
    return reversed;
}
console.log(reverseIntegerDigits(1234), reverseIntegerDigits(-1234), reverseIntegerDigits(123.4), reverseIntegerDigits(-123.4));

File: 5.8/kt/reverseDigits.kt
/*Kotlin gives us a name shaddow compiler warning because
we reassign the n value to itself. In this instance this is
an acceptable choice, thus supressing this warning */
@Suppress("NAME_SHADOWING") 
fun reverseDigits(n: Int): Int {
	var digit = n % 10
	var n = (n - digit) / 10
	var reversed = digit
	while (n != 0) {
		digit = n % 10
		n = n/ 10
		reversed = (reversed * 10) + digit
	}
	return reversed
}

fun main() {
	println(reverseDigits(1234))
	println(reverseDigits(-1234))
}
